packages/notte-sdk/src/notte_sdk/endpoints/agents.py

```
@@ -275,82 +288,157 @@ async def watch_logs(self, agent_id: str, session_id: str, log: bool = True) ->
+
+                            if should_stop:
+                                return None
+
+                            if response is not None and response.is_completed():
+                                logger.info(f"Agent {agent_id} completed in {counter} steps")
+                                return None
+
+                    except ConnectionError as e:
+                        logger.error(f"Connection error: {agent_id} {e}")
+                        return None
+                    except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)